lib/SILOptimizer/Transforms/SimplifyCFG.cpp

```diff
@@ -3202,7 +3203,10 @@ bool simplifyToSelectValue(SILBasicBlock *MergeBlock, unsigned ArgNum,
       auto *BrInst = cast<CondBranchInst>(CaseInfo.CmpOrDefault->getTerminator());
       if (FoundCmpBlocks.count(BrInst->getFalseBB()) != 1)
         return false;
-      Cases.push_back({CaseInfo.Literal, CaseInfo.Result});
+      // Ignore duplicate cases
+      if (CaseLiterals.insert(CaseInfo.Literal)) {
```

Comment:
You also must check if the Result value matches. E.g.
...
} else if input == 1 {
  result = Enum.A
} else  if input == 1 {
  result = Enum.B
} else ...

If it does not match you should bail. You could also check which value actually wins, but this is more tricky.